File: generator/on-board/src/main/scala/ru/mipt/acsl/geotarget/OnBoardModelRegistry.scala
package ru.mipt.acsl.geotarget

import ru.mipt.acsl.decode.model.registry.Registry
import ru.mipt.acsl.decode.parser.ModelRegistry

/**
  * @author Artem Shein
  */
object OnBoardModelRegistry {

  import ModelRegistry._

  val FoundationSource = SourceFileName("foundation")
  val FsSource = SourceFileName("fs")
  val IdentificationSource = SourceFileName("identification")
  val MccSource = SourceFileName("mcc")
  val PhotonSource = SourceFileName("photon")
  val ScriptingSource = SourceFileName("scripting")
  val SegmentationSource = SourceFileName("segmentation")
  val TmSource = SourceFileName("tm")
  val RoutingSource = SourceFileName("routing")

  val Sources = Seq(FoundationSource, FsSource, IdentificationSource, MccSource, PhotonSource, ScriptingSource,
    SegmentationSource, TmSource, RoutingSource) ++ ModelRegistry.Sources

  def registry: Registry = ModelRegistry.registryForSourceNames(Sources)

}

File: generator/on-board/src/main/java/ru/mipt/acsl/geotarget/OnBoardModelRegistry.java
package ru.mipt.acsl.geotarget;

import com.google.common.collect.Lists;
import ru.mipt.acsl.decode.model.registry.Registry;
import ru.mipt.acsl.decode.parser.ModelRegistry;
import ru.mipt.acsl.decode.parser.SourceFileName;

import java.util.List;

/**
 * Created by metadeus on 08.06.16.
 */
public class OnBoardModelRegistry {

    public static class Sources {

        public static final SourceFileName FOUNDATION = SourceFileName.newInstance("foundation");
        public static final SourceFileName FS = SourceFileName.newInstance("fs");
        public static final SourceFileName IDENTIFICATION = SourceFileName.newInstance("identification");
        public static final SourceFileName MCC = SourceFileName.newInstance("mcc");
        public static final SourceFileName PHOTON = SourceFileName.newInstance("photon");
        public static final SourceFileName SCRIPTING = SourceFileName.newInstance("scripting");
        public static final SourceFileName SEGMENTATION = SourceFileName.newInstance("segmentation");
        public static final SourceFileName TM = SourceFileName.newInstance("tm");
        public static final SourceFileName ROUTING = SourceFileName.newInstance("routing");

        public static final List<SourceFileName> ALL = Lists.newArrayList(FOUNDATION, FS,
                IDENTIFICATION, MCC, PHOTON, SCRIPTING, SEGMENTATION, TM, ROUTING);

        static {
            ALL.addAll(ModelRegistry.Sources.ALL);
        }

        // prevent from instantiation
        private Sources() {

        }

    }

    public static Registry registry() {
        return ModelRegistry.registryForSourceNames(Sources.ALL);
    }

    // prevent from instantiation
    private OnBoardModelRegistry() {

    }

}
